File: Partner-Api/src/main/java/work/slhaf/partner/api/factory/module/annotation/AgentModule.java
package work.slhaf.partner.api.factory.module.annotation;


import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 用于注解执行模块
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface AgentModule {

    /**
     * 模块名称
     */
    String name();

    /**
     * 模块执行顺序，数字越小执行越靠前
     */
    int order();
}

File: Partner-Framework/src/main/java/work/slhaf/partner/api/agent/factory/component/abstracts/AgentModule.kt
package work.slhaf.partner.api.agent.factory.component.abstracts

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import work.slhaf.partner.api.agent.factory.component.annotation.AgentComponent
import work.slhaf.partner.api.agent.runtime.interaction.flow.RunningFlowContext

/**
 * 模块基类
 */
@AgentComponent
sealed class AbstractAgentModule {

    var moduleName: String = javaClass.simpleName

    @JvmField
    val log: Logger = LoggerFactory.getLogger(javaClass.simpleName)

    abstract class Running<T : RunningFlowContext> : AbstractAgentModule() {

        abstract fun execute(context: T)

        abstract fun order(): Int
    }

    abstract class Sub<I, O> : AbstractAgentModule() {
        abstract fun execute(input: I): O
    }

    abstract class Standalone : AbstractAgentModule()

}

